docs/devcontainer/tutorial.ipynb

```diff
@@ -133,65 +146,166 @@
       "----------------------------------------------------------------------------------------\n",
-      "1          1    UNDEFINED    72.86     6.79    10.00   350.00  -300.00   300.00\n",
-      "2          1    UNDEFINED   114.07    -5.13    10.00   300.00  -300.00   300.00\n",
-      "3          1    UNDEFINED    80.21   -23.47    10.00   270.00  -300.00   300.00\n",
+      "1          1          NG    71.10    17.70    10.00   350.00  -300.00   300.00\n",
+      "2          1          NG   111.40    -2.61    10.00   300.00  -300.00   300.00\n",
+      "3          1        WIND    85.00   -13.42     0.00    85.00  -300.00   300.00\n",
       "[ExaGO] Finalizing opflow application.\n"
      ]
-    },
-    {
-     "data": {
-      "text/plain": [
-       "0"
-      ]
-     },
-     "execution_count": 3,
-     "metadata": {},
-     "output_type": "execute_result"
     }
    ],
    "source": [
     "exago.initialize(\"opflow\")\n",
     "opf = exago.OPFLOW()\n",
-    "opf.read_mat_power_data('datafiles/case9/case9mod.m')\n",
+    "opf.read_mat_power_data('datafiles/case9/case9mod_gen3_wind2.m')\n",
     "opf.solve()\n",
+    "filepath = './tmp.output'\n",
+    "opf.save_solution(exago.CSV, filepath)\n",
     "opf.print_solution()\n",
     "del opf\n",
     "exago.finalize()"
    ]
   },
   {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'nbus': [9.0], 'ngen': [3.0], 'nbranch': [9.0], 'baseMVA': [100.0], 'Bus': [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0], 'Pd': [0.0, 0.0, 0.0, 0.0, 75.0, 90.0, 0.0, 100.0, 0.0], 'Qd': [0.0, 0.0, 0.0, 0.0, 50.0, 30.0, 0.0, 35.0, 0.0], 'Vm': [1.04, 1.02, 1.02, 1.03, 1.01, 1.02, 1.03, 1.02, 1.03], 'Va': [0.0, 4.54, 2.9, -2.19, -3.38, -4.26, 0.75, -1.72, 0.21], 'mult_Pmis': [2064.25, 2013.83, 2018.37, 2064.55, 2076.02, 2093.63, 2014.5, 2035.99, 2018.82], 'mult_Qmis': [0.0, -0.0, -0.0, 0.78, 11.05, 5.44, 3.54, 7.25, 2.49], 'From': [1.0, 2.0, 3.0, 4.0, 4.0, 5.0, 6.0, 7.0, 8.0], 'To': [4.0, 7.0, 9.0, 5.0, 6.0, 7.0, 9.0, 8.0, 9.0], 'Status': [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0], 'Sft': [73.27, 111.43, 86.05, 32.2, 42.77, 49.98, 50.0, 63.9, 42.3], 'Stf': [72.63, 111.84, 86.79, 43.0, 44.87, 48.69, 51.77, 64.98, 36.62], 'Slim': [380.0, 250.0, 300.0, 250.0, 250.0, 250.0, 150.0, 250.0, 150.0], 'mult_Sf': [-0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0], 'mult_St': [-0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0, -0.0], 'Gen bus': [1.0, 2.0, 3.0], 'Status-2': [1.0, 1.0, 1.0], 'Fuel': ['NG', 'NG', 'WIND'], 'Pg': [71.1, 111.4, 85.0], 'Qg': [17.7, -2.61, -13.42], 'Pmin': [10.0, 10.0, 0.0], 'Pmax': [350.0, 300.0, 85.0], 'Qmin': [-300.0, -300.0, -300.0], 'Qmax': [300.0, 300.0, 300.0]}\n",
+      "data converted from ./tmp.output to ./data.json\n"
+     ]
+    }
+   ],
+   "source": [
+    "import csv \n",
```

Comment:
Can we put this source in a separate python file and then import the function. I think this source will be useful beyond the notebook.

Reply:
The idea is just to show users an example of how to roll their own, and then we can truly support this from the python wrapper in another PR with C++ code that generates an actual python API.